Library/Homebrew/cmd/update.sh

```diff
@@ -390,29 +390,23 @@ EOS
     fi
   fi
 
-  if [[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core/.git/shallow" ]]
-  then
-    odie <<EOS
-homebrew-core is a shallow clone. To \`brew update\` first run:
-  git -C "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core" fetch --unshallow
-This restriction has been made on GitHub's request because updating shallow
-clones is an extremely expensive operation due to the tree layout and traffic of
-Homebrew/homebrew-core. We don't do this for you automatically to avoid
-repeatedly performing an expensive unshallow operation in CI systems (which
-should instead be fixed to not use shallow clones). Sorry for the inconvenience!
-EOS
-  fi
+  [[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core/.git/shallow" ]] && SHALLOW_CORE=1
+  [[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-cask/.git/shallow" ]] && SHALLOW_CASK=1
```

Comment:
```suggestion
  [[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core/.git/shallow" ]] && HOMEBREW_CORE_SHALLOW=1
  [[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-cask/.git/shallow" ]] && HOMEBREW_CASK_SHALLOW =1
```

Reply:
Perhaps these could be set to the actual message e.g.

```bash
if [[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-cask/.git/shallow" ]]; then
  HOMEBREW_CASK_SHALLOW="$(cat <<EOS
homebrew-cask is a shallow clone
To \`brew update\`, first run:
  git -C "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-cask" fetch --unshallow
EOS
)"
fi
```

which could then be used below like

```
${HOMEBREW_CORE_SHALLOW}${HOMEBREW_CASK_SHALLOW}This restriction
```

Reply:
suggested line `HOMEBREW_CASK_SHALLOW =1` (note the space) is an error and should be corrected to `HOMEBREW_CASK_SHALLOW=1`

Reply:
thanks, @glensc. I've updated Mike's suggestion.

Reply:
I've changed all the variable names.